Cargo.lock

```diff
@@ -2808,6 +2859,21 @@ dependencies = [
  "uv-normalize",
 ]
 
+[[package]]
+name = "python-pkginfo"
+version = "0.6.3"
+source = "registry+https://github.com/rust-lang/crates.io-index"
+checksum = "ba3f3f0d552c7efdde2b6898bf21b49c4e76b3e6071ff196dfe52109804db896"
+dependencies = [
+ "flate2",
+ "fs-err",
+ "mailparse",
+ "rfc2047-decoder",
+ "tar",
+ "thiserror",
+ "zip",
+]
```

Comment:
These are some heavy dependencies. We otherwise don't depend on `tar` or `chumsky`. Can we find another way to do this part?

Reply:
Honestly, I'd prefer to just copy over this struct and put it in `uv-metadata`: https://github.com/PyO3/python-pkginfo-rs/blob/main/src/metadata.rs. We don't need any of these dependencies. We already have our own pattern for mailparse headers from `uv-metadata`.

Reply:
(I feel strongly on this one so if you disagree, let's discuss further rather than merging.)

Reply:
done